lib/appium.js

```diff
@@ -60,125 +150,23 @@ class AppiumDriver extends BaseDriver {
-        driverClass: WindowsDriver,
-        moduleName: 'appium-windows-driver',
-      },
-      MacDriver: {
-        driverClass: MacDriver,
-        moduleName: 'appium-mac-driver',
-      },
-      EspressoDriver: {
-        driverClass: EspressoDriver,
-        moduleName: 'appium-espresso-driver',
-        automationName: this.validAutomations.ESPRESSO,
-      },
-      TizenDriver: {
-        driverClass: TizenDriver,
-        moduleName: 'appium-tizen-driver',
-        automationName: this.validAutomations.TIZEN,
-      },
-    };
-    for (const driverProps of _.values(this._driversMapping)) {
-      driverProps.moduleInfo = require(`${driverProps.moduleName}/package.json`);
-    }
-
-    return this._driversMapping;
-  }
-
-  get platformsMapping () {
-    if (this._platformsMapping) {
-      return this._platformsMapping;
-    }
-
-    this._platformsMapping = {
-      fake: () => FakeDriver,
-      android: (caps) => {
-        const platformVersion = semver.valid(semver.coerce(caps.platformVersion));
-        if (platformVersion && semver.satisfies(platformVersion, '>=6.0.0')) {
-          log.warn("Consider setting 'automationName' capability to " +
-            `'${this.validAutomations.UIAUTOMATOR2.toLowerCase()}' ` +
-            "on Android >= 6, since UIAutomator framework " +
-            "is not maintained anymore by the OS vendor.");
-        }
-
-        return AndroidDriver;
-      },
-      ios: (caps) => {
-        const platformVersion = semver.valid(semver.coerce(caps.platformVersion));
-        if (platformVersion && semver.satisfies(platformVersion, '>=10.0.0')) {
-          log.info("Requested iOS support with version >= 10, " +
-            `using '${this.validAutomations.XCUITEST.toLowerCase()}' ` +
-            "driver instead of UIAutomation-based driver, since the " +
-            "latter is unsupported on iOS 10 and up.");
-          return XCUITestDriver;
-        }
-
-        return IosDriver;
-      },
-      windows: () => WindowsDriver,
-      mac: () => MacDriver,
-      tizen: () => TizenDriver,
-    };
-
-    return this._platformsMapping;
-  }
-
-  async getDriverForCaps (caps) {
+  getDriverForCaps (caps) {
     if (!_.isString(caps.platformName)) {
       throw new Error("You must include a platformName capability");
     }
 
     // we don't necessarily have an `automationName` capability,
     if (_.isString(caps.automationName)) {
-      for (const {automationName, driverClass} of _.values(this.driversMapping)) {
+      for (const {automationName, driverClass} of _.values(DRIVER_MAP)) {
         if (_.toLower(automationName) === caps.automationName.toLowerCase()) {
           return driverClass;
         }
       }
     }
 
-    const driverSelector = this.platformsMapping[caps.platformName.toLowerCase()];
+    const driverSelector = PLATFORMS_MAP[caps.platformName.toLowerCase()];
     if (driverSelector) {
-      return await driverSelector(caps);
+      return driverSelector(caps);
```

Comment:
I'd rather keep it sync just in case it will be necessary to add some more advanced driver selection algorithm